File: julia/perf/perf_ngsim_env.jl
using AutoEnvs

function perf_ngsim_env_step(n_steps=20000)
    filepath = Pkg.dir("NGSIM", "data", "trajdata_i101_trajectories-0750am-0805am.txt")
    params = Dict(
        "trajectory_filepaths"=>[filepath],
    )
    env = NGSIMEnv(params)
    action = [1.,0.]
    reset(env)
    @time for _ in 1:n_steps
        _, _, terminal, _ = step(env, action)
        if terminal
            reset(env)
        end
    end
end

function perf_vectorized_ngsim_env_step(n_steps=20000, n_envs=10)
    filepath = Pkg.dir("NGSIM", "data", "trajdata_i101_trajectories-0750am-0805am.txt")
    params = Dict(
        "trajectory_filepaths"=>[filepath],
    )
    env = VectorizedNGSIMEnv(n_envs, params)
    actions = zeros(2, n_envs)
    actions[1,:] = 1.
    reset(env)
    @time for _ in 1:n_envs:n_steps
        _, _, terminals, _ = AutoEnvs.step(env, actions)
        reset(env, terminals)
    end
end

perf_ngsim_env_step(20000)
perf_vectorized_ngsim_env_step(20000, 100)

File: python/perf/perf_ngsim_env.py

import numpy as np
import os

from julia_env.julia_env import JuliaEnv

from context_timer import ContextTimer

def perf_ngsim_env_step():
    basedir = os.path.expanduser('~/.julia/v0.6/NGSIM/data')
    filename = 'trajdata_i101_trajectories-0750am-0805am.txt'
    filepaths = [os.path.join(basedir, filename)]
    env = JuliaEnv(
        env_id='NGSIMEnv',
        env_params=dict(
            trajectory_filepaths=filepaths,
        ),
        using='AutoEnvs'
    )
    n_steps = 10000
    action = np.array([1.,0.])
    env.reset()
    with ContextTimer():
        for _ in range(n_steps):
            _, _, terminal, _ = env.step(action)
            if terminal:
                env.reset()

def perf_vectorized_ngsim_env_step():
    basedir = os.path.expanduser('~/.julia/v0.6/NGSIM/data')
    filename = 'trajdata_i101_trajectories-0750am-0805am.txt'
    filepaths = [os.path.join(basedir, filename)]
    n_envs = 100
    env = JuliaEnv(
        env_id='VectorizedNGSIMEnv',
        env_params=dict(
            n_envs=n_envs,
            trajectory_filepaths=filepaths,
        ),
        using='AutoEnvs'
    )
    n_steps = 10000
    action = np.zeros((n_envs, 2))
    action[:,0] = 1.
    env.reset()
    with ContextTimer():
        for _ in range(0, n_steps, n_envs):
            _, _, terminal, _ = env.step(action)
            env.reset(terminal)

if __name__ == '__main__':
    perf_ngsim_env_step()
    perf_vectorized_ngsim_env_step()